gemd/entity/object/process_run.py

```diff
@@ -121,6 +121,9 @@ def __eq__(self, other):
             if result is True and isinstance(other, ProcessRun):
                 if len(self.ingredients) == len(other.ingredients):
                     result = all(ing in other.ingredients for ing in self.ingredients)
+                elif (not self.ingredients and self.uids) \
+                        or (not other.ingredients and other.uids):
+                    result = True  # One can be empty if you flattened
```

Comment:
So they are equal if they have matching ingredients OR one of them doesn't have any ingredients defined?

Reply:
Mostly correct.  If one have them has no ingredients and there was a UID dictionary to validate.  

Intended to compensate for the fact that `ingredients` is set implicitly by a relationship that wouldn't be initialized if you just compared the result of `dataset.process_runs.get()` in citrine-python.

Reply:
Those aren't compared here - they are tested in `DictSerializable.__eq__`.  I'm intending to make sure that _any_ uid exists.  I would be open to changing that line to (len() == 0 / len() != 0) if that reads better.

Reply:
Yeah, that's what I meant.